themes/src/main/resources/theme/base/admin/resources/js/controllers/clients.js

```diff
@@ -1096,6 +1096,8 @@ module.controller('ClientDetailCtrl', function($scope, realm, client, templates,
             }
             $scope.client.publicClient = false;
             $scope.client.serviceAccountsEnabled = true;
+        } else if ($scope.client.bearerOnly) {
```

Comment:
This is actually makes sure that the service account is disabled once authz services are disabled. Its stays enabled before after deactivating them.